### ISSUE TITLE
Emoji From Other Servers in Message Text Often Fail to Render

### ISSUE THREAD

**Issue by Shadowmech88** [slatinsky/DiscordChatExporter-frontend#19]
I'm basically mirroring a report I made on the Discord Chat Exporter Github because I don't know of any way to tell whether this is an issue with the export/data or with the rendering/parsing of the data. Feel free to close this if you know it's out of your hands.

## Expected Behavior

All emoji display successfully.

## Actual Behavior

Emoji which are from other servers (besides the one which was exported) _and_ are posted in-line with the message text rather than as reactions, seem to very frequently show up in chat as :emojiname: rather than the image of the actual emoji.
Sometimes such emoji will display fine, no idea why. Sometimes one of them will display fine and an emoji right below it that's _from the exported server_ will fail to render. I've seen at least one instance of an emoji from another server rendering, but showing the wrong emoji when compared side-by-side with the live channel from which it was extracted.

## Steps to reproduce the problem

  1. Download section of Discord channel that has emoji which are from different servers posted in messages as part of the message text, as well as in reactions to messages.
  2. Open export in DCEF.
  3. Private emoji are shown as plaintext :emoji: while all other emoji show up fine, including the server-external emoji in reactions.

### Environment

  - Operating system: Windows 10
  - CPU Architecture: x86_64 (PC)
  - Flavor: Windows binary release

### DiscordChatExporter-frontend

  - [Release](https://github.com/slatinsky/DiscordChatExporter-frontend/releases) version or commit hash: v1.13.2
  - Can the issue be reproduced in the [demo](https://dcef.slada.sk/)?: Don't know

### [DiscordChatExporter](https://github.com/Tyrrrz/DiscordChatExporter)

- Version (CLI `--version`): v2.39.1

How did you export the data?
  - Export format (CLI `--format`): Json
  - Downloaded assets (CLI `--media`): Yes
  - Markdown processing (CLI `--markdown`): Yes

**Comment by slatinsky** [comment]
If markdown processing it turned on (default) in DiscordChatExporter, emoji ids are stripped from the export.

My frontend is finds all emojis used in reactions, and uses them to replace emojis in message content. Sometimes successfully, sometimes not (if the emoji was only used in message content and never in any reaction). 

If you turn off markdown processing in DiscordChatExporter `--markdown false` and enable internet access in DiscordChatExporter-frontend, emojis from other servers will render correctly. 

I can't fix this, sorry. It is missing from the export made by DiscordChatExporter.

**Comment by Shadowmech88** [comment]
What are the ramifications of not using markdown processing? The emoji not rendering are annoying, but would the export as a whole look much worse without markdown?

**Comment by slatinsky** [comment]
If exported without markdown processing, the export contains more raw information, like emojiIds, channelIds,...

**Comment by Shadowmech88** [comment]
I tried it out and that does work, although unfortunately I think the lack of readable user mentions and channel links is more crippling to the backup than broken emoji. I wonder how difficult it would be to make a tool to take two backups, one with markdown processing and one without, and create a new one where all markdown is processed _except_ for external emoji.

Also I noticed that without markdown processing, all in-message emoji are very large. I'm not sure if that's something you have control over with the frontend, though.

**Comment by slatinsky** [comment]
User mentions should be readable and highlighted. If they are not, it is a bug. 

But it is not possible to convert roles to their names, because that information is not exported by the exporter. 

Converting channel links to their names is possible, but not yet done (for exports exported using --markdown false).

If message only contains emoji and nothing else, emoji renders larger in discord too. That is intended behavior (issue #18)

**Comment by Shadowmech88** [comment]
> User mentions should be readable and highlighted

Without markdown processing on the export, they're not.
I pulled the same post up side by side and without markdown processing, the user mention is rendered in the style of `@1234567890` (presumably the user's ID number.)
Unless that's something which was fixed in newer DCEF versions, since I'm using v1.13.2.

> If message only contains emoji and nothing else, emoji renders larger in discord too

I know that. I'm actually the author of the issue you linked.
I'm talking about emoji rendering differently on exports for which markdown processing is turned off. Not only are they large even when the message also contains text, but they're much larger than even the large versions that Discord uses for emoji-only messages.
